src/lavinmq/queue/stream_queue_message_store.cr

```diff
@@ -49,12 +64,15 @@ module LavinMQ
         end
       end
 
-      private def offset_at(seg, pos) : Tuple(Int64, UInt32, UInt32)
+      private def offset_at(seg, pos, retried = false) : Tuple(Int64, UInt32, UInt32)
         return {@last_offset, seg, pos} if @size.zero?
         mfile = @segments[seg]
         msg = BytesMessage.from_bytes(mfile.to_slice + pos)
         offset = offset_from_headers(msg.properties.headers)
         {offset, seg, pos}
+      rescue ex : IndexError # first segment can be empty if message size >= segment size
+        return offset_at(seg + 1, pos, true) unless retried
```

Comment:
what if many segments has been deleted? this only tired one more segment? and should `pos` be reused? Shouldn't it be in the beginning of the next available segment? 
```suggestion
        return offset_at(@segments.first_key, 4, true) unless retried
```

Reply:
`offset_at()` is always called with `@segments.first_key` (or `@segments.last_key`), so `seg + 1` will always be the second segment (unless we're looking in the last segment, but since it's the last segment, we will never run into the issue of a message spilling over to the next segment in that case). 

And I don't think we ever will have two empty segments in a row? Anytime a message spills over, it will end up in the next segment. The way we expire segments for streams shouldn't allow for any gaps. But maybe getting the next key in the hash is safer? 

`pos `should be set to 4 though 👍